### DISCUSSION TITLE
Make the cache explorable

### DISCUSSION THREAD

**Discussion by manuphatak** [apollographql/apollo-client-devtools#867] [Ideas]
## Problem

Even in a small app, the cache can hold 1,000s to 100,000s records. Unfortunately, if you want to find a cache entry, you have to scroll through the list and find it visually with no tooling assist.  The typical browser "ctrl-f" search option is unavailable in chrome dev tools.   

(See screenshots at the end)

## Solution

Ideas:

1. Make the cache searchable.  

    Please give us the option to type our key into a search box and filter down the cache entries.
2. Turn object refs into clickable links

    Take this cache entry as an example: 
    <img width="627" alt="image" src="https://user-images.githubusercontent.com/5052422/166469827-4a2a70d2-c397-4787-874b-92f854cb66b0.png">

    I'd love to click into `Account:2f95e503-a874-4d5c-8f7b-d78e6d2fab73` and see what data we have for that cache entry.

## Screens

Imagine trying to find your ID in this list

![Screen Recording 2022-05-03 at 8 53 47 AM 2022-05-03 09_05_43](https://user-images.githubusercontent.com/5052422/166468448-a7dd125f-0323-43b5-be81-7859e513627c.gif) <img width="378" alt="image" src="https://user-images.githubusercontent.com/5052422/166467582-bf85c00e-db44-4e8f-879b-9d2367d90913.png">

**Comment by cgil** [comment]
🙏

**Comment by jerelmiller** [comment]
Hey @manuphatak 👋 !

This functionality was just merged (https://github.com/apollographql/apollo-client-devtools/pull/960) and should be published with the next release. We are hoping to get this out sometime soon. The search functionality is not yet improved, but at least you'll be able to navigate refs within the cache tab.

**Reply by manuphatak** [reply]
That's going to be huge; thank you, thank you, thank you!

**Comment by joeshub** [comment]
+1 for making the cache list filter by the search input value

**Comment by jerelmiller** [comment]
Hey all 👋 

Appreciate the patience on this request. I've got a PR open to address the search filtering (https://github.com/apollographql/apollo-client-devtools/pull/1056) which should provide a much improved experience. Look for that to be released hopefully sometime in the coming week!

**Reply by manuphatak** [reply]
Woohoo, looking forward to that extension update!! 🎉

**Comment by alessbell** [comment]
With https://github.com/apollographql/apollo-client-devtools/pull/1056 merged and released I'm going to go ahead and close out this discussion 🎉

Let me know if I should reopen it if there's anything I've missed. For new ideas feel free to create a separate discussion thread. Thanks all!
